src/AutoRest.CSharp/Mgmt/Models/MgmtClientOperation.cs

```diff
@@ -73,6 +73,7 @@ private MgmtClientOperation(IReadOnlyList<MgmtRestOperation> operations, Formatt
             _extensionParameter = extensionParameter;
             IdVariableName = idVariableName;
             IsConvenientOperation = isConvenientOperation;
+            MethodSignature = EnsureMethodSignature();
```

Comment:
well this is not what I meant and this might lead to issues since the properties used to construct the MethodSignature (such as `ReturnType`) are lazy loaded, to ensure everything is always working, it should lazy load.

I mean the new part you added could be determined in the ctor (the assignment of attributes).

Or it is OK that we make MethodSignature to be lazy loaded, since no other place will need those attributes.

Reply:
then i think it's better to have the init in one place than put them separated. I will revert the change back.